### PR TITLE
syslog/Kconfig: support config multi syslog channel

### PR DESCRIPTION
**Pull request by Donny9**
## Summary
We can select multi syslog channel(syslog ramlog、syslog console、syslog default and so on) by config instead of adding multi channel by API syslog_channel.
## Testing
daily test.

### CODE REVIEW

File: drivers/syslog/Kconfig [hunk 1]
```diff
@@ -215,20 +211,21 @@ config SYSLOG_RPMSG
...
 	---help---
 		syslog() interfaces will be present, but all output will go to the
 		up_putc(ARCH_LOWPUTC == y) or bit-bucket(ARCH_LOWPUTC == n).
 
-endchoice
+endif
+
+config SYSLOG_RPMSG_SERVER_NAME
+	string "The name of Syslog Rpmsg Server"
+	depends on SYSLOG_RPMSG
```
Comment (gustavonihei) [inline drivers/syslog/Kconfig:225]: Missing default value for string config

File: drivers/syslog/Kconfig [hunk 2]
```diff
@@ -195,7 +190,7 @@ config SYSLOG_CHAR
 
 config RAMLOG_SYSLOG
 	bool "Use RAMLOG for SYSLOG"
-	depends on RAMLOG && !ARCH_SYSLOG
+	depends on RAMLOG
```
Comment (gustavonihei) [inline drivers/syslog/Kconfig:193]: Missing default value for bool config

File: drivers/syslog/Kconfig (hunk 1 above)
Comment (xiaoxiang781216) [reply drivers/syslog/Kconfig:225]: There isn't a suitable default value, so it's better to no default value at all and then enforce the user must give one in defconfig.

File: drivers/syslog/Kconfig (hunk 2 above)
Comment (xiaoxiang781216) [reply drivers/syslog/Kconfig:193]: the default is false, so we can skip it safely.

Comment (gustavonihei) [reply drivers/syslog/Kconfig:193]: I believe it is better to be explicit with Kconfig default values.
This ways we ensure the consistency of our Kconfig use across the codebase. Furthermore, it is easier to understand when you see the default field.

Comment (xiaoxiang781216) [reply drivers/syslog/Kconfig:193]: Ok.

File: drivers/syslog/Kconfig (hunk 1 above)
Comment (gustavonihei) [reply drivers/syslog/Kconfig:225]: Ok, I agree.
Does Kconfig assume `""` as the default value for a string config?
This enforcement is currently not implemented, right?

File: drivers/syslog/Kconfig (hunk 2 above)
Comment (Donny9) [reply drivers/syslog/Kconfig:194]: Done!

File: drivers/syslog/Kconfig (hunk 1 above)
Comment (Donny9) [reply drivers/syslog/Kconfig:228]: Done!

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@gustavonihei do you have more comment?

**Comment by gustavonihei** [comment]
> @gustavonihei do you have more comment?

I think I'm okay with the current state. I saw that the string config default option was changed to `""`, but I haven't noticed the enforcement method. Did I miss something or is it okay to proceed this way?

**Comment by xiaoxiang781216** [comment]
> > @gustavonihei do you have more comment?
> 
> I think I'm okay with the current state. I saw that the string config default option was changed to `""`, but I haven't noticed the enforcement method. Did I miss something or is it okay to proceed this way?

We try with/without default "", the result is same(SYSLOG_RPMSG_SERVER_NAME=""), so it look like that there is no mehtod to enforce the user give a value in defconfig.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**